test-data/unit/cmdline.test

```diff
@@ -68,6 +81,7 @@ undef
 undef
 [out]
 pkg/a.py:1: error: Name 'undef' is not defined
+pkg/subdir/a.py:1: error: Name 'undef' is not defined
```

Comment:
Test importing from modules found via recursive directory traversal. Test cases where the directory contains `__init__.py` and where it doesn't exist.

Maybe test this also with `--namespace-packages`?

Reply:
testCmdlinePackageSlash tests when the directory contains __init__.py and testCmdlinePackageContainingSubdir tests when it doesn't.
I added importing of the modules.

Reply:
There currently isn't an interaction with `--namespace-packages` and cmdline tests are slow, so I won't do that unless told.
That said, I have another diff coming soon where I flesh out the package root code and make `--namespace-packages` the default. All these tests pass with that.

Reply:
I'll also update the docs once all these changes (essentially important bug fixes + first three points of #8584) are shipped :-)